src/Bicep.LangServer/BicepCompilationManager.cs

```diff
@@ -271,7 +271,13 @@ private CompilationContextBase CreateCompilationContext(IWorkspace workspace, Do
             }
         }
 
-        private (ImmutableArray<ISourceFile> added, ImmutableArray<ISourceFile> removed) UpdateCompilationInternal(DocumentUri documentUri, int? version, IDictionary<ISourceFile, ISemanticModel> modelLookup, IEnumerable<ISourceFile> removedFiles, bool triggeredByFileOpenEvent = false)
+        // TODO(asilverman): This signture is a bit awkward. We should consider refactoring to make it more clear what the inputs and outputs are.
```

Comment:
I don't think the input params will be much clear with a typedef.  removedFiles could use a comment.